app/src/main/java/com/example/drawingApp/DrawingViewModel.kt

```diff
@@ -1,22 +1,58 @@
 package com.example.drawingApp
 
-import android.graphics.Bitmap
 import android.graphics.Color
-import androidx.annotation.ColorInt
+import android.graphics.Color.HSVToColor
+import androidx.core.graphics.toColor
 import androidx.lifecycle.ViewModel
 
 class DrawingViewModel : ViewModel() {
 
-
-    @ColorInt
-    var primaryColor = Color.BLACK
-
-    val layerNames: MutableList<String> = mutableListOf()
-
-    var activeBitmap: Bitmap? = null
-
-    var isLayerCreationDialogOpen = false
-    var currentLayerCreationDialogText = ""
-
-
+    data class ViewState(
+        val isColorSheetOpen: Boolean
+    )
+
+    private var isColorPickerSheetOpen = false
+
+    var chosenColor: Color? = null
+
+    var hsvH = 0f
```

Comment:
who should the hsv data class belong to?
I'm almost thinking it should be its own standalone data class at this point rather than sitting in the util or the viewmodel.